src/app/api/v1/user/welcome-scan/create/route.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../../../utils/auth";
+import { NextRequest, NextResponse } from "next/server";
+
+import {
+  createProfile,
+  createScan,
+  isEligible,
+} from "../../../../../functions/server/onerep";
+import type { ProfileData } from "../../../../../functions/server/onerep";
+import AppConstants from "../../../../../../appConstants";
+import { getSubscriberByEmail } from "../../../../../../db/tables/subscribers";
+
+import {
+  setOnerepProfileId,
+  setOnerepScan,
+} from "../../../../../../db/tables/onerep_scans";
+import { setProfileDetails } from "../../../../../../db/tables/onerep_profiles";
+import { StateAbbr } from "../../../../../../utils/states";
+
+export interface WelcomeScanBody {
+  success: boolean;
+}
+
+export async function POST(
+  req: NextRequest
+): Promise<NextResponse<WelcomeScanBody | unknown>> {
+  const session = await getServerSession(authOptions);
+
+  const eligible = await isEligible();
+  if (!eligible) {
+    throw new Error("User has no more manual scans left");
+  }
+
+  const params = await req.json();
+  const requiredParams = [
+    "firstName",
+    "lastName",
+    "city",
+    "state",
+    "dateOfBirth",
+  ];
+  requiredParams.forEach((param) => {
+    if (!params[param]) {
+      throw new Error(`${param} is required`);
+    }
+  });
+
+  const { firstName, lastName, city, state, dateOfBirth } = params;
+  const profileData: ProfileData = {
+    first_name: firstName,
+    last_name: lastName,
+    addresses: [{ city: city as string, state: state as StateAbbr }],
+    birth_date: dateOfBirth,
+  };
+
+  if (typeof session?.user?.email === "string") {
+    try {
+      const subscriber = await getSubscriberByEmail(session.user.email);
+      // FIXME check if scan has already been performed and return early if so.
```

Comment:
@rhelmer Is this still something we need to fix?

Reply:
So it seems like we already sufficiently check this, so I removed the comment in [31a6f65](https://github.com/mozilla/blurts-server/pull/3222/commits/31a6f650df5673b5598881022f875d0ea625d83b).